src/main/java/com/exasol/bucketfs/UnsynchronizedBucket.java

```diff
@@ -40,13 +42,14 @@
      * BucketFS.
      * </p>
      *
-     * @param pathInBucket path inside the bucket
      * @param localPath    path of the file to be uploaded
+     * @param pathInBucket path inside the bucket
      * @throws TimeoutException      if the synchronization check takes too long
      * @throws BucketAccessException if the file cannot be uploaded to the given URI
      * @throws FileNotFoundException in case the source file is not found
+     * @return {@link UploadResult} describing the status of the upload
      */
-    void uploadFileNonBlocking(Path localPath, String pathInBucket)
+    UploadResult uploadFileNonBlocking(Path localPath, String pathInBucket)
```

Comment:
```suggestion
    public UploadResult uploadFileNonBlocking(Path localPath, String pathInBucket)
```

Reply:
it's an interface --> public by default. In this project the style seemed to be to not add public in that case.